### ISSUE TITLE
Could fairing run return the URL and other information?

### ISSUE THREAD

**Issue by jlewi** [kubeflow/fairing#212]
After building and deploying a model with fairing it might be useful to be able to access information.

For example, I'd like to know the URL of the endpoint so I can send predictions to it after calling fairing.config.run

https://github.com/kubeflow/fairing/blob/b69139ff86f1d87870858ddf56d6140f83fd21e6/fairing/config.py#L97

Would it make sense to add one or more URL methods to deployer so that I could do something like

```
fairing.config.run()
deployer = fairing.config.get_deployer()
deployer.get_url()
```

It might make sense to have multiple URLs or parameterize get_url so we could return different addresses e.g.

* InCluster DNS name
* External access point

**Comment by jlewi** [comment]
It looks like fairing_config.run doesn't store the preprocessor, builder, deployer anywhere.

**Comment by jlewi** [comment]
Maybe run could just return the preprocessor, builder, and deployer?

**Comment by jlewi** [comment]
Fixed by #207
